src/code.cloudfoundry.org/gorouter/route/pool.go

```diff
@@ -452,6 +497,23 @@ func (p *EndpointPool) Endpoints(logger *slog.Logger, initial string, mustBeStic
 	}
 }
 
+func (p *EndpointPool) FallBackToDefaultLoadBalancing(defaultLBAlgo string, logger *slog.Logger, initial string, mustBeSticky bool, azPreference string, az string) EndpointIterator {
+	logger.Info("hash-based-routing-header-not-found",
+		slog.String("poolLBAlgorithm", p.LoadBalancingAlgorithm),
+		slog.String("Host", p.host),
+		slog.String("Path", p.contextPath))
+
+	switch defaultLBAlgo {
+	case config.LOAD_BALANCE_LC:
+		logger.Debug("endpoint-iterator-with-least-connection-lb-algo")
+		return NewLeastConnection(logger, p, initial, mustBeSticky, azPreference == config.AZ_PREF_LOCAL, az)
```

Comment:
if you pass in `azPreference` as boolean, like for the other LB algos, you don't need to repeat the condition in-line 3 times.